nixos/modules/services/networking/syncthing.nix

```diff
@@ -51,79 +55,82 @@ let
             --retry 1000 --retry-delay 1 --retry-all-errors \
             "$@"
     }
-  '' +
-
-  /* Syncthing's rest API for the folders and devices is almost identical.
-  Hence we iterate them using lib.pipe and generate shell commands for both at
-  the sime time. */
-  (lib.pipe {
-    # The attributes below are the only ones that are different for devices /
-    # folders.
-    devs = {
-      new_conf_IDs = map (v: v.id) devices;
-      GET_IdAttrName = "deviceID";
-      override = cfg.overrideDevices;
-      conf = devices;
-      baseAddress = "${cfg.guiAddress}/rest/config/devices";
-    };
-    dirs = {
-      new_conf_IDs = map (v: v.id) folders;
-      GET_IdAttrName = "id";
-      override = cfg.overrideFolders;
-      conf = folders;
-      baseAddress = "${cfg.guiAddress}/rest/config/folders";
-    };
-  } [
-    # Now for each of these attributes, write the curl commands that are
-    # identical to both folders and devices.
-    (mapAttrs (conf_type: s:
-      # We iterate the `conf` list now, and run a curl -X POST command for each, that
-      # should update that device/folder only.
-      lib.pipe s.conf [
-        # Quoting https://docs.syncthing.net/rest/config.html:
-        #
-        # > PUT takes an array and POST a single object. In both cases if a
-        # given folder/device already exists, it’s replaced, otherwise a new
-        # one is added.
-        #
-        # What's not documented, is that using PUT will remove objects that
-        # don't exist in the array given. That's why we use here `POST`, and
-        # only if s.override == true then we DELETE the relevant folders
-        # afterwards.
-        (map (new_cfg: ''
-          curl -d ${lib.escapeShellArg (builtins.toJSON new_cfg)} -X POST ${s.baseAddress}
-        ''))
+
+    ${if (cfg.overrideDevices && cfg.overrideFolders) then ''
+      # If we need to override both devices and folders, we can just use a single PUT request to push the new config
+      curl -X PUT ${cfg.guiAddress}/rest/config -d ${lib.escapeShellArg (builtins.toJSON newConfig)}
+    '' else
```

Comment:
I don't like how the behavior of this script changes fundamentally when someone changes these `override{Folder,Device}s`...